src/audio/google/google_rtc_audio_processing.c

```diff
@@ -546,13 +551,36 @@ static int google_rtc_audio_processing_prepare(struct processing_module *mod,
 		return -EINVAL;
 	}
 
+	/* check IBS/OBS in streams */
+	if (cd->num_frames * source_get_frame_bytes(sources[cd->raw_microphone_source]) !=
+	    source_get_min_available(sources[cd->raw_microphone_source])) {
+		comp_warn(dev, "Incorrect IBS on microphone source: %d, expected %u",
+			  source_get_min_available(sources[cd->raw_microphone_source]),
+			  cd->num_frames *
+				 source_get_frame_bytes(sources[cd->raw_microphone_source]));
+	}
+	if (cd->num_frames * sink_get_frame_bytes(sinks[0]) !=
+	    sink_get_min_free_space(sinks[0])) {
+		comp_warn(dev, "Incorrect OBS on sink :%d, expected %u",
+			  sink_get_min_free_space(sinks[0]),
+			  cd->num_frames * sink_get_frame_bytes(sinks[0]));
+	}
+	if (cd->num_frames * source_get_frame_bytes(sources[cd->aec_reference_source]) !=
+	    source_get_min_available(sources[cd->aec_reference_source])) {
+		comp_warn(dev, "Incorrect IBS on reference source: %d, expected %u",
+			  source_get_min_available(sources[cd->aec_reference_source]),
+			  cd->num_frames *
+				source_get_frame_bytes(sources[cd->aec_reference_source]));
+	}
```

Comment:
a bit strange - we say that inputs are wrong, but we don't fail and don't fix them?

Reply:
yes, long story. 
As description says, the PR is not completely finished